backend/tests/test_post_api.py

```diff
@@ -0,0 +1,44 @@
+from django.contrib.auth import get_user_model
+from backend.models import Post
+
+import pytest
+
+User = get_user_model()
+test_user_username = "testuser001"
+test_user_email = "testemail001@gmail.com"
+test_user_password = "ualberta!"
+test_user_github_url = "https://github.com/testuser001"
+
+
+@pytest.mark.django_db
+class TestPostAPI:
+
+    @pytest.fixture
+    def test_user(self, db):
+        test_user = User.objects.create_user(
+            username=test_user_username, email=test_user_email, password=test_user_password, githubUrl=test_user_github_url)
+        return test_user
+
+    def test_get_post_by_id(self, client, test_user):
+        test_post = Post.objects.create(author=test_user, title="post title", content="post content")
```

Comment:
So currently we need to pass in the user object to create a post? Can we make it so that it’s only the user ID we need to pass or do we need other information from the user object?

Reply:
You don't need to pass in anything related to user to create a post, it will automatically bind to currently authenticated user  as only logged in user can create a post.  I can update the test later to become closer to real usage example.